streamer/src/nonblocking/connection_rate_limiter.rs

```diff
@@ -1,156 +1,85 @@
 use {
-    governor::{DefaultDirectRateLimiter, DefaultKeyedRateLimiter, Quota, RateLimiter},
-    std::{net::IpAddr, num::NonZeroU32},
+    solana_net_utils::token_bucket::{KeyedRateLimiter, TokenBucket},
+    std::net::IpAddr,
 };
 
 /// Limits the rate of connections per IP address.
 pub struct ConnectionRateLimiter {
-    limiter: DefaultKeyedRateLimiter<IpAddr>,
+    limiter: KeyedRateLimiter<IpAddr>,
 }
 
+/// The threshold of the size of the connection rate limiter map. When
+/// the map size is above this, we will trigger a cleanup of older
+/// entries used by past requests.
+const CONNECTION_RATE_LIMITER_CLEANUP_SIZE_THRESHOLD: usize = 100_000;
+
 impl ConnectionRateLimiter {
     /// Create a new rate limiter per IpAddr. The rate is specified as the count per minute to allow for
-    /// less frequent connections.
+    /// less frequent connections. Higher limit also allows higher bursts.
     pub fn new(limit_per_minute: u64) -> Self {
-        let quota =
-            Quota::per_minute(NonZeroU32::new(u32::try_from(limit_per_minute).unwrap()).unwrap());
         Self {
-            limiter: DefaultKeyedRateLimiter::keyed(quota),
+            limiter: KeyedRateLimiter::new(
+                CONNECTION_RATE_LIMITER_CLEANUP_SIZE_THRESHOLD,
+                TokenBucket::new(
+                    limit_per_minute,
+                    limit_per_minute,
+                    limit_per_minute as f64 / 60.0,
+                ),
+                8,
+            ),
         }
     }
 
     /// Check if the connection from the said `ip` is allowed.
+    /// Here we assume that only IPs with actual confirmed connections are stored in it,
+    /// since we should only modify server state once source IP is verifired
```

Comment:
nits: verified

Reply:
fixed in https://github.com/anza-xyz/agave/commit/dd66eae309f73e5ac3a477145a325a2e51b0558b